From 70e4fb30f84a53050039902ef1acad9f84c45e64 Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Wed, 22 May 2019 13:51:51 +0800
Subject: [PATCH] Add the third argument for the open()

When O_CREAT or O_TMPFILE is in the second argument, the third argument
is mandatory. Otherwise it will trigger build errors like follow:
In function 'open', inlined from 'write_buf_to_file' at lib/file_utils.c:108:7:
  rc-19.06.0-r0/recipe-sysroot/usr/include/bits/fcntl2.h:50:4: error: call to '__open_missing_mode' declared with attribute error: open with O_CREAT or O_TMPFILE in second argument needs 3 arguments
      __open_missing_mode ();
      ^~~~~~~~~~~~~~~~~~~~~~

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 apps/common/mvapp.c  | 3 ++-
 src/lib/file_utils.c | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/apps/common/mvapp.c b/apps/common/mvapp.c
index 94674b9f2f44..017e0bcdcab2 100644
--- a/apps/common/mvapp.c
+++ b/apps/common/mvapp.c
@@ -237,7 +237,8 @@ static int print_to_file_cb(const char *fmt, ...)
 		printf("%s: buffer overflow (%d chars)\n", __func__, n);
 	va_end(ap);
 
-	fd = open(mvapp->cli_out_filename, O_CREAT | O_WRONLY | O_APPEND);
+	fd = open(mvapp->cli_out_filename, O_CREAT | O_WRONLY | O_APPEND,
+		  S_IRWXU | S_IRGRP);
 	if (fd <= 0) {
 		pr_err("can't open CLI file (%s)\n", mvapp->cli_out_filename);
 		return -EIO;
diff --git a/src/lib/file_utils.c b/src/lib/file_utils.c
index e5d243478517..b2d675aeb4e0 100644
--- a/src/lib/file_utils.c
+++ b/src/lib/file_utils.c
@@ -105,7 +105,7 @@ int write_buf_to_file(char *file_name, char *buff, u32 size)
 	size_t	s;
 	int	fd;
 
-	fd = open(file_name, O_RDWR | O_CREAT);
+	fd = open(file_name, O_RDWR | O_CREAT, S_IRWXU | S_IRGRP);
 	if (fd == -1) {
 		pr_err("Failed to open file %s\n", file_name);
 		return -EIO;
-- 
2.14.4

